torch/utils/__init__.py

```diff
@@ -46,6 +46,32 @@ def swap_attr(name):
         setattr(t1, name, (getattr(t2, name)))
         setattr(t2, name, tmp)
 
+    def error_pre_hook(grad_outputs):
+        raise RuntimeError("Trying to execute AccumulateGrad node that was poisoned by swap_tensors "
+                           "this can happen when you try to run backward on a tensor that was swapped. "
+                           "For a module m with `torch.__future__.set_swap_module_params_on_conversion(True)` "
+                           "this could happen if trying to run backward changing the device or dtype of the module "
```

Comment:
```suggestion
                           "this could happen if trying to run backward after changing the device or dtype of the module "
```

Reply:
actually.. maybe I would frame it as 
1. changing device/dtype after running forward (and then running backward) is not allowed
2. please change device/dtype before running forward

because if you tell me to swap .cpu() with backward, the changes don't take effect unless the user also wanted to run a second iteration?